src/WorkBC.Data/Migrations/20240413000723_InsertNocCodes2021.cs

```diff
@@ -0,0 +1,136 @@
+                }
+                else
+                {
+                    try
+                    {
+                        int.TryParse(nocs.noc_2021, out int id);
+
+                        string code = nocs.noc_2021.ToString();
+                        string title = nocs.label_en.ToString();
+                        string frenchTitle = nocs.label_fr.ToString();
+
+                        //Console.WriteLine("id: " + id);
+                        //Console.WriteLine("code: " + code);
+                        //Console.WriteLine("title: " + title);
+                        //Console.WriteLine("frenchTitle: " + frenchTitle);
+
+                        migrationBuilder.InsertData(
+                            table: "NocCodes2021",
+                            columns: new[] { "Id", "Code", "Title", "FrenchTitle" },
+                            values: new object[] { id, code, title, frenchTitle });
+
+                    }
+                    catch (Exception ex)
+                    {
+                        Console.WriteLine("Inside catch in up: " + ex.Message.ToString());
+                    }
+                }
+
+
+            }
+
+        }
+
+        protected async override void Down(MigrationBuilder migrationBuilder)
+        {
+            Console.WriteLine("Inside Insert Down method");
+            //Truncate table- delete all records
+            migrationBuilder.Sql(@"TRUNCATE TABLE [dbo].[NocCodes2021]");
+        }
+        public async Task<List<NocCodeSSOT>> GetNocCodes2021(string nocCodes2021SSOTUrl)
+        {
+
+            List<NocCodeSSOT> newSSOTCodes = new List<NocCodeSSOT>();
+            try
+            {
+                using (var _httpClient = new HttpClient())
+                {
+                    var response = await _httpClient.GetAsync(nocCodes2021SSOTUrl);
+                    if (!response.IsSuccessStatusCode)
+                    {
+                        throw new Exception("NO SSOT response: " + response.Content);
+                    }
+                    if (response.IsSuccessStatusCode)
+                    {
+                        var jsonString = await response.Content.ReadAsStringAsync();
+                        Console.WriteLine("Inside GetNoc method");
+
+                        // convert string to stream
+                        byte[] byteArray = Encoding.UTF8.GetBytes(jsonString);
+                        //byte[] byteArray = Encoding.ASCII.GetBytes(contents);
+                        MemoryStream jsonStream = new MemoryStream(byteArray);
+                        Console.WriteLine("After stream method");
+                        newSSOTCodes = await JsonSerializer.DeserializeAsync<List<NocCodeSSOT>>(jsonStream);
+
+                        Console.WriteLine("After deserialize method");
+                    }
+
+                }
+
+
+            }
+            catch (Exception ex)
+            {
+                throw new Exception("Inside catch" + ex.Message.ToString());
```

Comment:
Same comment as previously - no need for `try .. catch` structure.

Reply:
Removed the try-catch blocks in the latest commit.